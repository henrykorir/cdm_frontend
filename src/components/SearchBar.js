import React, { useState, useEffect } from 'react'
import InputField from './InputField'
/*
const initialState = null
const reducer = (state, action) =>{
	switch(action.type){
		case 'SEARCH_PATIENT_BY_NAME':
			return {
				...state, rows:[...state.rows, action.payload]
			}
		default:
			return state;
	}
}
*/
const SearchBar = ({ getPatient, setAllPatients }) => {
	const [ patientsData, setPatientsData ] = useState(null)
	const [ patientName, setPatientName ] = useState("")
	
	const getPatientName = ( name ) => {
		setPatientName(name)
	}
	useEffect(() => {
		fetch('https://ampath.herokuapp.com/patients')
		.then( result => result.json())
		.then( data => {console.log(data);setPatientsData(data); setAllPatients(data)})
		.catch(error => console.error(error))
	})
	
	useEffect(() => {
		const searchPatientByName = ( name ) => {
			if( patientsData ){
				const { rows } = patientsData
				const patientRecord = rows.filter( row => row['PatientName'] === patientName)
				console.log(patientRecord)
				getPatient(patientRecord)
			}
		}
		searchPatientByName(patientName)
	},[ patientName, patientsData, getPatient])
	
	return(
		<div className="flex justify-center bg-green-200 py-4">
			<InputField onGetPatientName = { getPatientName }/>
		</div>
	)
}
export default SearchBar