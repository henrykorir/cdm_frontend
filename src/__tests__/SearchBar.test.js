import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";

import SearchBar from '../components/SearchBar'

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
}); 

describe('SearchBar', () => {
	it('should render correctly',() =>{
		act(() =>{
			render(<SearchBar />, container);
		})
		expect(container.innerHTML).toMatchSnapshot();
	})
	it('should render with text and a callback', () => {
		const onChange = jest.fn();
		act(() =>{
			render(<SearchBar text={""} onChange={ onChange } />, container);
		});
		//expect(text).not.toBeDefined();
		expect(onChange).toBeDefined();
	});
});
