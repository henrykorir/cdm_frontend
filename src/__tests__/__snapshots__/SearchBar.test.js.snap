// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render correctly 1`] = `"<div class=\\"flex justify-center bg-gray-800 py-4\\"><input class=\\"border-2 focus:border-blue-500 hover:border-blue-500 w-6/12 h-8 text-center\\" type=\\"text\\" placeholder=\\"Enter patient name to search ...\\" value=\\"\\"></div>"`;
